### PR TITLE
Make the discovery test more strict

### LINKED ISSUE #1645: Bandwidth usage is ludicrously high
Full nodes on Alexander are reporting 1-3MiB/s regularly. This should be more in the range of 25-100Kib/s

### PR DESCRIPTION
**Pull request by tomaka**
Close #1645 

We now check for the bandwidth usage during discovery (no more than 1kiB/sec/node), and make sure to properly account for disconnects (which can happen in case of simultaneous dialing).

### CODE REVIEW

File: core/network-libp2p/tests/test.rs [hunk 1]
```diff
@@ -141,18 +141,23 @@ fn many_nodes_connectivity() {
 			let mut num_connecs = 0;
 			stream::poll_fn(move || -> io::Result<_> {
 				loop {
+					const MAX_BANDWIDTH: u64 = NUM_NODES as u64 * 1024;		// 1kiB/s/node
+					assert!(node.average_download_per_sec() < MAX_BANDWIDTH);
```
Comment (gavofyork) [inline core/network-libp2p/tests/test.rs]: prove or remove!

File: core/network-libp2p/tests/test.rs [hunk 2]
```diff
@@ -141,18 +141,23 @@ fn many_nodes_connectivity() {
...
+					assert!(node.average_upload_per_sec() < MAX_BANDWIDTH);
```
Comment (gavofyork) [inline core/network-libp2p/tests/test.rs]: ditto

File: core/network-libp2p/tests/test.rs [hunk 3]
```diff
@@ -141,18 +141,23 @@ fn many_nodes_connectivity() {
...
+
 					match try_ready!(node.poll()) {
 						Some(ServiceEvent::OpenedCustomProtocol { .. }) => {
 							num_connecs += 1;
+							assert!(num_connecs < NUM_NODES);
```
Comment (gavofyork) [inline core/network-libp2p/tests/test.rs]: ditto

**Comment by tomaka** [comment]
I was assuming that we didn't have to prove panics within tests?
I can't really prove that the bandwidth will be lower than 1kiB/sec/node.

**Comment by bkchr** [comment]
I think this was a misunderstanding of @gavofyork and you are right that panics in tests are allowed.

**Pull request merged**